hare/hare.go

```diff
@@ -193,18 +193,23 @@ var ErrTooLate = errors.New("consensus process finished too late")
 
 // records the provided output.
 func (h *Hare) collectOutput(ctx context.Context, output TerminationOutput) error {
-	set := output.Set()
-	blocks := make([]types.BlockID, len(set.values))
-	i := 0
-	for v := range set.values {
-		blocks[i] = v
-		i++
+	layerID := output.ID()
+	var blocks []types.BlockID
+	if output.Completed() {
+		h.WithContext(ctx).With().Info("hare terminated with success", layerID)
+		set := output.Set()
+		blocks = make([]types.BlockID, len(set.values))
+		i := 0
+		for v := range set.values {
+			blocks[i] = v
```

Comment:
will be cleaner with capacity instead of len and append

Reply:
done